crypto/keyring/keyring.go

```diff
@@ -467,29 +467,48 @@ func wrapKeyNotFound(err error, msg string) error {
 }
 
 func (ks keystore) List() ([]Info, error) {
-	var res []Info
+	res := []Info{}
 
 	keys, err := ks.db.Keys()
 	if err != nil {
 		return nil, err
 	}
 
-	sort.Strings(keys)
+	if len(keys) == 0 {
+		return res, nil
+	}
 
+	sort.Strings(keys)
 	for _, key := range keys {
 		if strings.HasSuffix(key, infoSuffix) {
 			rawInfo, err := ks.db.Get(key)
 			if err != nil {
-				return nil, err
+				fmt.Printf("err for key %s: %q\n", key, err)
+
+				// add the name of the key in case the user wants to retrieve it
+				// afterwards
+				info := newOfflineInfo(key, nil, hd.PubKeyType(""))
+				res = append(res, info)
+				continue
 			}
 
 			if len(rawInfo.Data) == 0 {
-				return nil, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key)
+				fmt.Println(sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, key))
```

Comment:
why are these prints changed? I don't think the same issue that the pr solves is being solved here

Reply:
say you have 100 keys in `os`. If you reject the password prompt, then there won't be any keys printed in even if you cancel when you approved 99/100